### ISSUE TITLE
Still maintained?

### ISSUE THREAD

**Issue by jimmywarting** [Leonidas-from-XIV/node-xml2js#626]
Are you looking for maintainers?
there is quite a bit of issues/pr open

**Comment by Leonidas-from-XIV** [comment]
So I don't have much time for this project. I am kinda unhappy about the current state of it (and the absolute nightmare of maintainability and consistency) so my plan has been to rewrite it (while also ditching CoffeeScript — it was useful while it lasted but now things have changed and CoffeeScript fell by the wayside). Hence I am really hesitant about taking in new features because the redesign will probably dump them altogether.

But when am I gonna have the time to properly maintain it? Who knows. My personal life TODO list is giant and doesn't show any sign of reducing.

**Comment by jimmywarting** [comment]
Ditching CoffeeScript would be 👍 #620
Would be nice to replace it with vanilla js and do a major release
switching to esm would be nice to so it can be imported from Deno, Browser & Nodes new http loader without any transpilation or need for NPM

**Comment by Leonidas-from-XIV** [comment]
No, removing the CoffeeScript source wouldn't do much, it is still a mess with way too high cyclomatic complexity and the output format is still essentially `???`. Hence why I would like to translate the functionality into a typed language, so users of the library would know what they are getting out and it wouldn't vary depending on what they feed in.

**Comment by cainthebest** [comment]
I think the even though the output is essentially `???` moving to a typed lang would be best for it as you cannot know the output but you could know the type of the output. Since it's already compiled into JavaScript just change over to typescript it may take a massive update and probably will have issues on the way but it would resolve what your looking for.

**Comment by suterma** [comment]
Just my 2 cents: I am using the library, and the only real drawback IMHO is that I can not control which nodes are handled as arrays and which ones are not. Since I need to map the data to an internal typed store, having typed output actually would not be much of an advantage in my scenario. I think this would also apply to many other users.

**Comment by jimmywarting** [comment]
I actually prefer plain javascript over TS
You can have both typed support and still use plain javascript and still be able to import stuff without the need for compilers or use of npm if you just wish to import it from Deno or Browsers without bundling

<details>
  <summary>how?</summary>

With JSDoc, checkJS + allowJS
```json
{
  "include": ["*.js"],
  "compilerOptions": {
    "target": "ES2020",
    "lib": ["ES2020", "DOM"],
    "module": "ES2020",
    "moduleResolution": "node",
    "allowJs": true,
    "checkJs": true,
    "declaration": true,
    "emitDeclarationOnly": true,
    "allowSyntheticDefaultImports": true,
    "skipLibCheck": true
  }
}
```

</details>

**Comment by tnrich** [comment]
I'm gonna hop in here and say that this project is no longer maintained. 

If anyone has any more up to date alternatives I would love to hear them!

**Comment by tnrich** [comment]
After quite a bit of searching I found what appears to be a fairly close and seemingly much better maintained replacement: 

https://github.com/NaturalIntelligence/fast-xml-parser

Here's how to use it in a similar but slightly different way from xml2js:

```js
import { XMLParser } from "fast-xml-parser";
new XMLParser({
          ignoreAttributes: false,
          attributeNamePrefix: "",
          isArray: (name) => name === "Feature" || name === 'Segment', 
        }).parse(xmlString);
```
